omigami/ms2deepscore.py

```diff
@@ -19,127 +13,45 @@ class InvalidCredentials(Exception):
     pass
 
 
-class InvalidNumberOfSpectra(Exception):
-    pass
-
-
-class MS2DeepScore:
+class MS2DeepScore(Endpoint):
     _endpoint_url = "https://omigami.datarevenue.com/seldon/seldon/ms2deepscore/api/v0.1/predictions"
 
     def __init__(self, token: str):
-        self._token = token
+        super().__init__(token)
 
-    def predict_similarity_of_pair(
+    def match_spectra_from_path(
         self,
         mgf_path: str,
+        n_best: int,
+        include_metadata: List[str] = None,
+        **kwargs,
     ) -> float:
         """
-        Predicts molecular structural similarities (Tanimoto scores) from pairs of
-        mass spectrometry spectra in local mgf files using MS2DeepScore algorithm
+        Finds the N best matches for spectra in a local mgf file using MS2DeepScore
+        algorithm.
 
         Parameters
         ----------
         mgf_path: str
             Local path to mgf file with two spectra
+        n_best: int
+            Number of best matches to select
+        include_metadata: List[str]
+            Metadata keys to include in the response. Will make response slower. Please
+            check the documentation for a list of valid keys.
 
         Returns
         -------
-        Molecular structural similarities (Tanimoto scores)
+        A list of pandas dataframes containing the best matches and optionally metadata
+        for these matches.
 
         """
-        spectra_generator = load_from_mgf(mgf_path)
-
-        payload = self._build_payload(spectra_generator)
-        response = self._send_request(payload)
-
-        prediction = self._format_results(response)
-        return prediction
-
-    def _build_payload(
-        self,
-        pair: Generator[Spectrum, None, None],
-    ) -> JSON:
-        """Build the json payload
+        parameters = self._build_parameters(n_best, include_metadata)
```

Comment:
choose endpoint considering the ion mode then (like in spec2vec). maybe it can be moved to the parent class